packages/blade-mcp/src/tools/getBladeComponentDocs.ts

```diff
@@ -91,6 +147,54 @@ const getBladeComponentDocsToolCallback: ToolCallback<typeof getBladeComponentDo
   }
 };
 
+// Callback for stdio transport
+const getBladeComponentDocsStdioCallback: ToolCallback<typeof getBladeComponentDocsStdioSchema> = ({
+  componentsList,
+  currentProjectRootDirectory,
+}) => {
+  return getBladeComponentDocsCore({
+    componentsList,
+    currentProjectRootDirectory,
+    skipLocalCursorRuleChecks: false, // Perform cursor rule checks for stdio
+    cursorRuleVersion: '0',
+    clientName: 'unknown',
+  });
+};
+
+// Callback for HTTP transport
+const getBladeComponentDocsHttpCallback: ToolCallback<typeof getBladeComponentDocsHttpSchema> = ({
+  componentsList,
+  cursorRuleVersion,
+  clientName,
+  currentProjectRootDirectory,
+}) => {
+  return getBladeComponentDocsCore({
+    componentsList,
+    currentProjectRootDirectory,
+    skipLocalCursorRuleChecks: true, // Skip cursor rule checks for HTTP
+    cursorRuleVersion,
+    clientName,
+  });
+};
+
+const getBladeComponentDocsToolSchema = (
+  transportType: 'stdio' | 'http',
+): typeof getBladeComponentDocsStdioSchema | typeof getBladeComponentDocsHttpSchema => {
+  if (transportType === 'stdio') {
+    return getBladeComponentDocsStdioSchema;
+  }
+  return getBladeComponentDocsHttpSchema;
+};
+
+const getBladeComponentDocsToolCallback = (
+  transportType: 'stdio' | 'http',
+): typeof getBladeComponentDocsStdioCallback | typeof getBladeComponentDocsHttpCallback => {
+  if (transportType === 'stdio') {
+    return getBladeComponentDocsStdioCallback;
+  }
+  return getBladeComponentDocsHttpCallback;
+};
```

Comment:
This much nesting in every function seems confusing. Would defining different tool calls in server.ts on check won't be a better approach?